src/main/java/team9/ddang/walk/service/WalkServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+    private final MemberRepository memberRepository;
+
+
+    @Override
+    @Transactional
+    public CompleteWalkAloneResponse completeWalk(Long memberId, CompleteWalkServiceRequest completeWalkServiceRequest) {
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow();
+        // TODO : security로 멤버 받아올 예정
+        Dog dog = getDogFromMemberId(memberId);
+        List<Location> locations = getLocationList(member.getEmail());
+
+        if(locations.isEmpty()) {
+            throw new IllegalArgumentException("산책이 정상적으로 이루어지지 않았습니다.");
+        }
+
+        Walk walk = completeWalkServiceRequest.toEntity(locations, member);
+        saveWalkAndLocationAndDog(locations, walk, dog);
+        removeMemberLocation(member.getEmail());
+
+        return CompleteWalkAloneResponse.of(
+                member.getName(), walk.getTotalDistance(), completeWalkServiceRequest.totalWalkTime(),
+                calculateCalorie(dog.getWeight(),completeWalkServiceRequest.totalDistance()),
+                locations.stream().map(Location::getPosition).toList()
+        );
+    }
+
+    private void saveWalkAndLocationAndDog(List<Location> locations, Walk walk, Dog dog){
+        locations.forEach(location -> location.updateWalk(walk));
+
+        WalkDog walkDog = WalkDog.builder()
+                .dog(dog)
+                .walk(walk)
+                .build();
+
+        walkRepository.save(walk);
+        walkDogRepository.save(walkDog);
+        locationBulkRepository.saveAll(locations);
+    }
+
+    private List<String> getListFromRedis(String email) {
+        String key = LIST_KEY + email;
+        ListOperations<String, String> listOperations = redisTemplate.opsForList();
+        List<String> locations = listOperations.range(key, 0, -1);
+
+        redisTemplate.delete(key);
+        return locations;
+    }
+
+    private List<Location> getLocationList(String email){
+        List<String> locations = getListFromRedis(email);
+
+        return locations != null && !locations.isEmpty() ?
+                locations.stream().map(this::getLocationTempFromString).toList() : Collections.emptyList();
+    }
+
+    private Location getLocationTempFromString(String location){
+        String[] parts = location.split(",");
+        double longitude = Double.parseDouble(parts[0].split("=")[1]);
+        double latitude = Double.parseDouble(parts[1].split("=")[1]);
+        LocalDateTime timeStamp = LocalDateTime.parse(parts[2].split("=")[1]);
+
+        return Location.createTemp(longitude, latitude, timeStamp);
+    }
+
+    private Dog getDogFromMemberId(Long memberId){
+        MemberDog memberDog = memberDogRepository.findMemberDogByMemberId(memberId)
+                .orElseThrow(() -> new IllegalArgumentException("멤버가 소유하고 있는 개가 없습니다!"));
+        return memberDog.getDog();
+    }
+
+    private void removeMemberLocation(String email){
+        redisTemplate.opsForGeo().remove(POINT_KEY, email);
+    }
```

Comment:
저는 데이터를 삭제할 때, 삭제 여부를 확인하는 편 입니다.
한번 고려해보세요.

```suggestion
    private void removeMemberLocation(String email){
        if (redisTemplate.opsForGeo().remove(POINT_KEY, email) == 0) {
    throw new IllegalStateException("위치 정보를 삭제하지 못했습니다.");
}
    }
```

Reply:
아하 넵 참고하겠습니다